### ISSUE TITLE
local @tied_array broken

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#4611]
Migrated from [rt.perl.org#7938](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=7938) (status was 'resolved')

Searchable as RT7938$

**Comment by p5pRT** [comment]
### From @schwern



Tied arrays cannot be localized\.  The expected output of this program
is&#8203;:

1 2 3 | 1 2 3
4 5 6 | 4 5 6
1 2 3 | 1 2 3

instead one gets

1 2 3 | 1 2 3
4 5 6 | 4 5 6
1 2 3 | 4 5 6

indicating the tied array is not restored to its orignal condition\.

&nbsp;    use Tie&#8203;::Array;
&nbsp;    tie @&#8203;ta\, "Tie&#8203;::StdArray";

&nbsp;    @&#8203;ta = \(1\,2\,3\);
&nbsp;    @&#8203;a  = \(1\,2\,3\);
&nbsp;    print "@&#8203;a | @&#8203;ta\\n";

&nbsp;    \{ 
&nbsp;        local @&#8203;ta = \(4\,5\,6\);
&nbsp;        local @&#8203;a  = \(4\,5\,6\);
&nbsp;        print "@&#8203;a | @&#8203;ta\\n";
&nbsp;    \}

&nbsp;    print "@&#8203;a | @&#8203;ta\\n";

This effects all perls from bleadperl back to at least 5\.005\_03\.


If it can't be fixed prior to 5\.8\.0\, it would be nice if C\<local
@&#8203;tied\_array> at least emitted some sort of warning\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

Site configuration information for perl v5.6.1:

Configured by bod at Fri Jun 22 19:29:41 EST 2001.

Summary of my perl5 (revision 5.0 version 6 subversion 1) configuration:
  Platform:
    osname=linux, osvers=2.4.5-pre3, archname=powerpc-linux
    uname='linux vir 2.4.5-pre3 #1 mon jun 18 15:58:49 est 2001 ppc unknown '
    config_args='-Dccflags=-DDEBIAN -Dcccdlflags=-fPIC -Darchname=powerpc-linux -Dprefix=/usr -Dprivlib=/usr/share/perl/5.6.1 -Darchlib=/usr/lib/perl/5.6.1 -Dvendorprefix=/usr -Dvendorlib=/usr/share/perl5 -Dvendorarch=/usr/lib/perl5 -Dsiteprefix=/usr/local -Dsitelib=/usr/local/share/perl/5.6.1 -Dsitearch=/usr/local/lib/perl/5.6.1 -Dman1dir=/usr/share/man/man1 -Dman3dir=/usr/share/man/man3 -Dman1ext=1 -Dman3ext=3perl -Dpager=/usr/bin/sensible-pager -Uafs -Ud_csh -Uusesfio -Dotherlibdirs=/usr/lib/perl5/5.6:/usr/lib/perl5/5.005 -Duseshrplib -Dlibperl=libperl.so.5.6.1 -Dd_dosuid -des'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
  Compiler:
    cc='cc', ccflags ='-DDEBIAN -fno-strict-aliasing -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2',
    cppflags='-DDEBIAN -fno-strict-aliasing -I/usr/local/include'
    ccversion='', gccversion='2.95.4 20010604 (Debian prerelease)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=4321
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=8
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, usemymalloc=n, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lgdbm -ldbm -ldb -ldl -lm -lc -lcrypt
    perllibs=-ldl -lm -lc -lcrypt
    libc=/lib/libc-2.2.3.so, so=so, useshrplib=true, libperl=libperl.so.5.6.1
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fPIC', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.6.1:
    /usr/local/lib/perl/5.6.1
    /usr/local/share/perl/5.6.1
    /usr/lib/perl5
    /usr/share/perl5
    /usr/lib/perl/5.6.1
    /usr/share/perl/5.6.1
    /usr/local/lib/site_perl/powerpc-linux
    /usr/local/lib/site_perl
    /usr/local/share/perl/5.6.0
    /usr/lib/perl5/5.6/powerpc-linux
    /usr/lib/perl5/5.6
    /usr/lib/perl5/5.005/powerpc-linux
    /usr/lib/perl5/5.005
    .


Environment for perl v5.6.1:
    HOME=/home/schwern
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/schwern/bin:/home/schwern/sbin:/usr/local/sbin:/usr/sbin:/sbin:/usr/local/bin:/usr/bin:/bin:/usr/local/X11/bin:/usr/bin/X11:/usr/games
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @schwern

Still an issue in 5\.8\.1 RC2\.

**Comment by p5pRT** [comment]
### From @iabyn

Now fixed in blead with commit 5afa72af5aa99c40932771ad390abf5ba229611b\.

See also
http&#8203;://www\.xray\.mpe\.mpg\.de/mailing\-lists/perl5\-porters/2010\-04/msg00561\.html

**Comment by p5pRT** [comment]
@iabyn - Status changed from 'open' to 'resolved'
